### PR TITLE
chore: Add validation to test release codebuild spec

### PR DESCRIPTION
**Pull request by farleyb-amazon**
*Description of changes:*

Now the test release validation does the same checks as the prod release (running the sample application against the new version), but targeting testpypi. To support this I've also refactored out the validation steps into a dedicated spec so both the prod and test specs can depend on it.

*Testing:*
Prod and test releases succeed (without any version uploaded, because it hasn't changed) when I give them the correct version and commit.

When I give a bad commit, we fail to checkout the branch and stop.

When I give a good commit but bad version, we detect that the authoritative version in `identifiers.py` does not match the requested version and stop.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: codebuild/release/prod-release.yml [hunk 1]
```diff
@@ -9,21 +9,33 @@ env:
...
+      - pip install tox
+      - pip install --upgrade pip
     runtime-versions:
       python: latest
+  pre_build:
+    commands:
+      - git checkout $COMMIT_ID
+      - FOUND_VERSION=$(sed -n 's/__version__ = "\(.*\)"/\1/p' src/aws_encryption_sdk/identifiers.py)
+      - |
+        if expr ${FOUND_VERSION} != ${VERSION}; then
```
Comment (robin-aws) [inline codebuild/release/prod-release.yml:22]: I'm assuming `VERSION` has to be provided manually when executing this build? Is there a good place to document a repository's available build commands and how to call them?

File: codebuild/release/test-release.yml [hunk 2]
```diff
@@ -9,17 +9,35 @@ env:
...
       - tox -e park
       - tox -e test-release
 
 
 batch:
-  fast-fail: false
-  build-list:
+  fast-fail: true
+  build-graph:
     - identifier: test_release
```
Comment (robin-aws) [inline codebuild/release/test-release.yml:35]: These labels are a little confusing: in isolation I'm not sure if this one means "test the release" or "release the test artifact". Perhaps we can use phrases like "release to production" vs "release to staging"?

File: codebuild/release/prod-release.yml [hunk 3]
```diff
@@ -9,21 +9,33 @@ env:
 
 phases:
   install:
+    commands:
+      - pip install tox
+      - pip install --upgrade pip
     runtime-versions:
       python: latest
+  pre_build:
```
Comment (robin-aws) [inline codebuild/release/prod-release.yml:17]: What's the functional difference between `build` and `pre_build`?

**Review by robin-aws — CHANGES_REQUESTED** [review]

File: codebuild/release/test-release.yml (hunk 2 above)
Comment (farleyb-amazon) [reply codebuild/release/test-release.yml:35]: Makes sense, will update wording.

File: codebuild/release/prod-release.yml (hunk 3 above)
Comment (farleyb-amazon) [reply codebuild/release/prod-release.yml:17]: To be honest I haven't found a clear description of any functional differences. The docs provide some examples of what you could do in each one, but there doesn't seem to be any concrete. My grouping here is fairly arbitrary, so I'm open to suggestions.

File: codebuild/release/prod-release.yml (hunk 1 above)
Comment (farleyb-amazon) [reply codebuild/release/prod-release.yml:22]: Correct, and good question. There is a spot for environment variables, but it looks like they do require you pass a value, and I didn't want to provide a default for the version. Worst case we could add comments, but let me look a bit more to see if there are other options.

File: codebuild/release/prod-release.yml (hunk 3 above)
Comment (robin-aws) [reply codebuild/release/prod-release.yml:17]: This is fine, it looks like `pre_build/build/post_build` are really just to help organize the phases, and this seems to fit well.

**Review by robin-aws — APPROVED** [review]

**Pull request merged**